src/Validation/ScannableURLProvider.php

```diff
@@ -110,12 +110,29 @@ public function get_urls() {
 		/*
 		 * If 'Your homepage displays' is set to 'Your latest posts', include the homepage.
 		 */
-		if ( 'posts' === get_option( 'show_on_front' ) && $this->is_template_supported( 'is_home' ) ) {
-			$urls[] = [
-				'url'   => home_url( '/' ),
-				'type'  => 'is_home',
-				'label' => __( 'Homepage', 'amp' ),
-			];
+		if ( 'posts' === get_option( 'show_on_front' ) ) {
+			if ( $this->is_template_supported( 'is_home' ) ) {
+				$urls[] = [
+					'url'   => home_url( '/' ),
+					'type'  => 'is_home',
+					'label' => __( 'Homepage', 'amp' ),
+				];
+			}
+		} elseif ( 'page' === get_option( 'show_on_front' ) ) {
+			if ( $this->is_template_supported( 'is_front_page' ) && get_option( 'page_on_front' ) ) {
+				$urls[] = [
+					'url'   => get_permalink( get_option( 'page_on_front' ) ),
+					'type'  => 'is_front_page',
+					'label' => __( 'Homepage', 'amp' ),
+				];
+			}
+			if ( $this->is_template_supported( 'is_home' ) && get_option( 'page_for_posts' ) ) {
+				$urls[] = [
+					'url'   => get_permalink( get_option( 'page_for_posts' ) ),
+					'type'  => 'is_home',
+					'label' => __( 'Blog', 'amp' ),
+				];
+			}
```

Comment:
When in legacy Reader mode, these should only be included only if AMP has been explicitly enabled for the page per this logic in `amp_is_available()`:

https://github.com/ampproject/amp-wp/blob/4cd08b2982682b5641c8376694f5f5ad414ee636/includes/amp-helper-functions.php#L541-L549

Reply:
I'm going to add some tests for this.

Reply:
Done in 59db0ed35